deployment/helm/node-feature-discovery/values.yaml

```diff
@@ -10,7 +10,8 @@ nameOverride: ""
 fullnameOverride: ""
 namespaceOverride: ""
 
-enableNodeFeatureApi: true
```

Comment:
I think we don't necessarily need to remove this. We could do a favor (and break some existing users) by adding a small snippet in `deployment/helm/node-feature-discovery/templates/_helpers.tpl` that sets `featureGates.NodeFeatureApi` based on this.

We should change the default, though


```
enableNodeFeatureApi: null
```

and only act if `enableNodeFeatureApi` is non-null.

WDYT?